### PR TITLE
Fix for infinite loop issue from network failure

### PR DESCRIPTION
**Pull request by Fizz11**
I have experienced a rare issue with the open connection loop listening for events when a sudden network outage occurs. 

I have observed the client.register() method continue to run long after the DCOS master node suddenly loses network connection for an extended period of time, and does not re-establish connection when the master comes back online. this results in any app using this API to basically be stuck until it is manually terminated or restarted. 

I have been able to reliably reproduce this by running test.py, and after successfully connecting and entering its loop, physically unplugging the network cable from our DCOS master node and keeping it disconnected for at least 3 minutes. 

The fix I found works is adding a connection timeout to the post request used to listen for mesos messages. In order to inadvertently change the behavior of this app I also added this timeout as an optional property that the user would have to set. Otherwise, behavior remains unchanged.

### CODE REVIEW

File: mesoshttp/client.py [hunk 1]
```diff
@@ -340,6 +340,8 @@ def __init__(
         :type frameworkWebUI: str
         :param max_reconnect: number of reconnection retries when connection fails
         :type max_reconnect: int  defaults to 3
+        :param connection_timeout: sets a timeout for scheduler connection loop
```
Comment (osallou) [inline mesoshttp/client.py:344]: You define param connection_timeout here, but it is not in def params if init.
You could add param with a default value to None.
Type/param are here to define input params.

If you only want to declare instance var, to be set by app, then just remove type/param def here.

Comment (Fizz11) [reply mesoshttp/client.py:344]: Nice catch! thanks for pointing that out, i have added connection_timeout as an input parameter for init.

**Comment by osallou** [comment]
Merged @Fizz11, thanks for analysis of this use case. Will to create a new release

**Comment by osallou** [comment]
@Fizz11 version 0.3.1 tag and pypi package includes your contribution

**Pull request merged**